coordinator/src/services/translation/incoming/release/additional-instructions.ts

```diff
@@ -0,0 +1,96 @@
+import * as fhir from "../../../../models/fhir/fhir-resources"
+import * as uuid from "uuid"
+import {createReference} from "../fhir-base-types"
+
+const MEDICATION_TAG_MATCHER = /^<medication>(.*?)<\/medication>/
+const PATIENT_INFO_TAG_MATCHER = /^<patientInfo>(.*?)<\/patientInfo>/
+const CONTROLLED_DRUG_PREFIX = "CD: "
+
+export function parseAdditionalInstructions(additionalInstructionsText: string): {
+  medication: Array<string>,
+  patientInfo: Array<string>,
+  controlledDrugWords: string,
+  additionalInstructions: string
+} {
+  const medication: Array<string> = []
+  const patientInfo: Array<string> = []
+
+  let medicationMatch = MEDICATION_TAG_MATCHER.exec(additionalInstructionsText)
+  let patientInfoMatch = PATIENT_INFO_TAG_MATCHER.exec(additionalInstructionsText)
+  while (medicationMatch || patientInfoMatch) {
+    if (medicationMatch) {
+      medication.push(medicationMatch[1])
+      additionalInstructionsText = additionalInstructionsText.substring(medicationMatch[0].length)
+    }
+    if (patientInfoMatch) {
```

Comment:
This could just be an else, but I don't think it would make a big difference :shrug: